src/OrchardCore.Modules/OrchardCore.ContentFields/Controllers/UserPickerAdminController.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using OrchardCore.Admin;
+using OrchardCore.ContentFields.Settings;
+using OrchardCore.ContentFields.Services;
+using OrchardCore.ContentFields.ViewModels;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.Modules;
+
+namespace OrchardCore.ContentFields.Controllers
+{
+    [RequireFeatures("OrchardCore.Users")]
+    [Admin]
+    public class UserPickerAdminController : Controller
+    {
+        private readonly IContentDefinitionManager _contentDefinitionManager;
+        private readonly IEnumerable<IUserPickerResultProvider> _resultProviders;
+
+        public UserPickerAdminController(
+            IContentDefinitionManager contentDefinitionManager,
+            IEnumerable<IUserPickerResultProvider> resultProviders
+            )
+        {
+            _contentDefinitionManager = contentDefinitionManager;
+            _resultProviders = resultProviders;
+        }
+
+        public async Task<IActionResult> SearchUsers(string part, string field, string query)
```

Comment:
This is using the same pattern as the ContentPicker. I was wondering if there was a way to secure this controller with proper permissions ?

Reply:
what would the proper permissions be?

We have `ManageUsers` but we have nothing for `DisplayUsers`

Reply:
I am not sure to be honest. Just pointing it out. Do you think this could be a security issue where someone who has the visit admin permission be able to get a list of  users ?

Reply:
We might need a `ViewUsers` permission. 

Because this is related to content editing, I'm not sure. It might be better here to restrict this access to Edit Content.

As you couldn't use it, unless you were editing a content item.

Reply:
This is the kind of field where it should only allow to search by UserName.

Reply:
I made it extensible @Skrypt so that someone could implement a search by email, or any of the other custom properties that they may have relating to users.

Like for example they might have a first name / last name to search, so they could search for Jasmin, rather than having to search for @Skrypt 

@jptissot I think I will tie the permission to `EditContent` for that content type, rather than a new permission for users or for this field.

Reply:
Makes sense! Thanks @deanmarcussen

Reply:
Permissions done

Reply:
LGTM